pkg/puller/puller.go

```diff
@@ -354,76 +251,68 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 		s, _, _, err := p.nextPeerInterval(peer, bin)
 		if err != nil {
 			p.metrics.HistWorkerErrCounter.Inc()
-			p.logger.Debug("histSyncWorker nextPeerInterval failed", "error", err)
+			p.logger.Error(err, "histSyncWorker nextPeerInterval failed")
 			return
 		}
+
 		if s > cur {
-			loggerV2.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur)
+			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur)
 			return
 		}
-		top, ruid, err := p.syncer.SyncInterval(ctx, peer, bin, s, cur)
+		top, _, err := p.syncer.SyncInterval(ctx, peer, bin, s, cur)
 		if err != nil {
-			loggerV2.Debug("histSyncWorker syncing interval failed", "peer_address", peer, "bin", bin, "cursor", cur, "error", err)
-			if ruid == 0 {
-				p.metrics.HistWorkerErrCounter.Inc()
-			}
-
-			// since we use bin context cancellation to cancel interval
-			// sync operations, the context can be expired here, causing us
-			// to try to send a message with an expired context, which is
-			// bound to fail.
-			ctxC, cancelC := context.WithTimeout(context.Background(), 10*time.Second)
-			defer cancelC()
-			if err := p.syncer.CancelRuid(ctxC, peer, ruid); err != nil {
```

Comment:
Why is this being removed? This makes it a breaking change in the protocols.

@janos Can you add some brief description about this cancel ruid functionality? Maybe we are missing something?

Reply:
The whole pullsync cancel protocol is a way to signal a call to a context's cancel function for a particular syncing request. To avoid unnecessary wait in pullsync handler if the "client" syncing peer has problems storing intervals. Since stream termination (reset) can be detected only on their Read or Write methods, it was needed to have a mechanism to terminate other functions that happen in between if there is an error on the other peer.